projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6481,16 +6480,14 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
             const overlaySettings: OverlaySettings = {
                 outlet: this.loadingOutlet,
                 closeOnOutsideClick: false,
-                positionStrategy: new ContainerPositionStrategy()
+                positionStrategy: new ContainerPositionStrategy(),
             };
-            if (!this._loadingId) {
-                this._loadingId = this.overlayService.attach(this.loadingOverlay, overlaySettings);
-                this.overlayService.show(this._loadingId, overlaySettings);
+            if (this.loadingOverlay.collapsed) {
+                this.loadingOverlay.open(overlaySettings);
             }
         } else {
-            if (this._loadingId) {
-                this.overlayService.hide(this._loadingId);
-                this._loadingId = null;
+            if (!this.loadingOverlay.collapsed) {
+                this.loadingOverlay.close();
             }
```

Comment:
There is no need to check `collapsed` property. It is checked internally in the toggle methods.